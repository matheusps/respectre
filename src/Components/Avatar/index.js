import React from 'react'
import { string, oneOf } from 'prop-types'

const Avatar = ({ size, background, image, chars }) => (
    <figure
    className={`avatar avatar-${size} bg-${background}`}
    data-initial={chars}>
        { !!image && <img src={image} alt="Avatar"></img> }
    </figure>
)

Avatar.defaultProps = {
    size: 'sm',
    background: 'primary'
}

Avatar.propTypes = {
    size: oneOf([ 'xl', 'lg', 'sm', 'xs' ]),
    background: oneOf([ 
        'primary',
        'secondary',
        'dark',
        'gray',
        'success',
        'warning',
        'error'
    ]),
    image: string,
    chars: string
}

export default Avatar