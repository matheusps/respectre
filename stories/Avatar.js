import React, { Fragment } from 'react'
import { storiesOf } from '@storybook/react'
import { Avatar } from '../lib/respectre'
import '../lib/respectre.css'

const EX_URL = 'https://picturepan2.github.io/spectre/img/avatar-1.png'

storiesOf('Avatar', module)
  .add('with text', () => (
    <Fragment>
      <Avatar size="xs" chars="XS" />
      <Avatar size="sm" chars="SM" />
      <Avatar size="lg" chars="LG" />
      <Avatar size="xl" chars="XL" />
    </Fragment>
  ))
  .add('with image', () => (
    <Fragment>
      <Avatar size="xs" image={EX_URL} />
      <Avatar size="sm" image={EX_URL} />
      <Avatar size="lg" image={EX_URL} />
      <Avatar size="xl" image={EX_URL} />
    </Fragment>
  ))
