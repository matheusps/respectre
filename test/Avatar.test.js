import React from 'react'
import Avatar from '../src/Components/Avatar'
import renderer from 'react-test-renderer'

describe('Avatar', () => {
  it('should match snapshot', () => {
    const component = renderer.create(<Avatar size="xl" chars="XL" />).toJSON()
    expect(component).toMatchSnapshot()
  })

  it('should render image correctly', () => {
    const withImage = renderer
      .create(<Avatar size="xl" image="testing" />)
      .toJSON()
    const withoutImage = renderer
      .create(<Avatar size="xl" chars="XL" />)
      .toJSON()

    expect(withImage.children).toHaveLength(1)
    expect(withImage.children[0].type).toEqual('img')
    expect(withoutImage.children).toBeNull
  })
})
