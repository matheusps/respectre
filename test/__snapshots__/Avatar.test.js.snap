// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should match snapshot 1`] = `
<figure
  className="avatar avatar-xl bg-primary"
  data-initial="XL"
/>
`;
